### PR TITLE
gpui: Add `opacity` to support transparency of the entire element

### PR DESCRIPTION
**Pull request by huacnlee**
Release Notes:

- N/A

---

Add this for let GPUI element to support fade in-out animation.

## Platform test

- [x] macOS
- [x] blade `cargo run -p gpui --example opacity --features macos-blade`

## Usage

```rs
div()
    .opacity(0.5)
    .bg(gpui::black())
    .text_color(gpui::black())
    .child("Hello world")
```

This will apply the `opacity` it self and all children to use `opacity` value to render colors.

## Example

```
cargo run -p gpui --example opacity
cargo run -p gpui --example opacity --features macos-blade
```

image

### CODE REVIEW

File: crates/gpui/src/elements/div.rs [hunk 1]
```diff
@@ -1500,35 +1500,44 @@ impl Interactivity {
...
-                    cx.with_text_style(style.text_style().cloned(), |cx| {
-                        cx.with_content_mask(style.overflow_mask(bounds, cx.rem_size()), |cx| {
-                            if let Some(hitbox) = hitbox {
-                                #[cfg(debug_assertions)]
-                                self.paint_debug_info(global_id, hitbox, &style, cx);
-
-                                if !cx.has_active_drag() {
-                                    if let Some(mouse_cursor) = style.mouse_cursor {
-                                        cx.set_cursor_style(mouse_cursor, hitbox);
+                cx.with_element_opacity(style.opacity, |cx| {
```
Comment (huacnlee) [inline crates/gpui/src/elements/div.rs:1503]: Here just wrap all code in `cx.with_element_opacity`

**Comment by huacnlee** [comment]
```
thread 'main' panicked at /Users/jason/.cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/fee06c4/blade-graphics/src/shader.rs:72:9:
assertion `left == right` failed: Host struct 'PolychromeSprite' size doesn't match the shader
  left: 92
 right: 96
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
libc++abi: terminating due to uncaught foreign exception
[1]    49412 abort      cargo run -p gpui --example opacity --features macos-blade
```

**Comment by huacnlee** [comment]
> ```
> thread 'main' panicked at /Users/jason/.cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/fee06c4/blade-graphics/src/shader.rs:72:9:
> assertion `left == right` failed: Host struct 'PolychromeSprite' size doesn't match the shader
>   left: 92
>  right: 96
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> libc++abi: terminating due to uncaught foreign exception
> [1]    49412 abort      cargo run -p gpui --example opacity --features macos-blade
> ```

The reason is need to add:

```rs
pub pad: u32, // align to 8 bytes
```

**Comment by huacnlee** [comment]
https://github.com/user-attachments/assets/ca1af63f-ef91-47ef-9cd3-ab00b247476a

**Comment by mrnugget** [comment]
Looks good to me! I tested it locally and went through the code. Very thorough contribution. Appreciate the example and the screenshots :)

**Pull request merged**